framework/src/vectorpostprocessors/ElementsAlongLine.C

```diff
@@ -47,9 +47,8 @@ ElementsAlongLine::execute()
   std::vector<Elem *> intersected_elems;
   std::vector<LineSegment> segments;
 
-  MooseSharedPointer<PointLocatorBase> plb = MooseSharedPointer<PointLocatorBase>(_fe_problem.mesh().getMesh().sub_point_locator().release());
-
-  Moose::elementsIntersectedByLine(_start, _end, _fe_problem.mesh(), plb, intersected_elems, segments);
+  UniquePtr<PointLocatorBase> pl = _fe_problem.mesh().getPointLocator();
+  Moose::elementsIntersectedByLine(_start, _end, _fe_problem.mesh(), pl, intersected_elems, segments);
```

Comment:
`elementsIntersectedByLine` never takes ownership of the pointer, IMO the reference is the right thing for passing this argument. Didn't we have that discussion a few weeks back?